tests/framework/web/RequestTest.php

```diff
@@ -741,4 +741,16 @@ public function testRequestMethodCanNotBeDowngraded($requestMethod, $requestOver
 
         $this->assertSame($expectedMethod, $request->getMethod());
     }
+
+    public function testTrustedHostAndSecureConnection()
+    {
+        $_SERVER['REMOTE_ADDR'] = '30.0.0.1';
```

Comment:
Try setting `REMOTE_ADDR` to client IP (50.0.0.1) and you will see that it does not work as expected.

Reply:
```
 Client   --->  Load Balancer  --->  Yii2 App
50.0.0.1          30.0.0.1           10.0.0.1
```

The `REMOTE_ADDR` cannot be `50.0.0.1`.

Reply:
You are wrong, because it's totally possible - that's my real case. If you want to reproduce it then you can use _load balancer_ for _nginx_ servers with `ngx_http_realip` module.

Reply:
That's what I mean, that's exactly what PR added.